### ISSUE TITLE
Optionally rename security-sensitive Redis functions

### ISSUE THREAD

**Issue by mogul** [18F/kubernetes-broker#80]
**Story:** In order to reduce the possibility that a compromised application will be able to destroy the content of a Redis instance, customers should be able to specify an optional prefix string that will be used to rename sensitive commands when they run `cf [create-service|update-service]` with the Redis broker.

**Acceptance Criteria:**

* GIVEN I am logged into cloud.gov \
  AND I [create a Redis instance](https://cloud.gov/docs/services/redis/#how-to-create-an-instance) using the command `cf create-service redis32 standard-ha my-redis-service
-c '{"admin_command_prefix": "<prefix>"}' ` \
  AND I [connect to that Redis instance using the `redis-cli` via port-forwarding](https://cloud.gov/docs/services/redis/#connecting-to-your-redis-service-locally) \
  WHEN I run `<command>` \
  THEN I see `<result>`
* GIVEN I am logged into cloud.gov \
  AND I create a Redis instance using the command `cf create-service redis32 standard-ha my-redis-service` \
  AND I update the Redis instance using the command `cf update-service my-redis-service
-c '{"admin_command_prefix": "<prefix>"}' ` \
  AND I [connect to that Redis instance using the `redis-cli` via port-forwarding](https://cloud.gov/docs/services/redis/#connecting-to-your-redis-service-locally) \
  WHEN I run `<command>` \
  THEN I see `<result>`

Examples:

| prefix | command | result |
| --- | --- | --- |
| rutabaga | BGREWRITEAOF | (error) ERR unknown command  |
| rutabaga | BGSAVE | (error) ERR unknown command  |
| rutabaga | CONFIG | (error) ERR unknown command  |
| rutabaga | DEBUG | (error) ERR unknown command  |
| rutabaga | DEL | (error) ERR unknown command  |
| rutabaga | FLUSHALL | (error) ERR unknown command  |
| rutabaga | FLUSHDB | (error) ERR unknown command  |
| rutabaga | KEYS | (error) ERR unknown command  |
| rutabaga | MONITOR | (error) ERR unknown command  |
| rutabaga | PEXPIRE | (error) ERR unknown command  |
| rutabaga | RENAME | (error) ERR unknown command  |
| rutabaga | SAVE | (error) ERR unknown command  |
| rutabaga | SHUTDOWN | (error) ERR unknown command  |
| rutabaga | SPOP | (error) ERR unknown command  |
| rutabaga | SREM | (error) ERR unknown command  |
| rutabaga | SWAPDB | (error) ERR unknown command  |
|  | BGREWRITEAOF | [something indicating a recognized command]  |
|  | BGSAVE | [something indicating a recognized command]  |
|  | CONFIG | [something indicating a recognized command]  |
|  | DEBUG | [something indicating a recognized command]  |
|  | DEL | [something indicating a recognized command]  |
|  | FLUSHALL | [something indicating a recognized command]  |
|  | FLUSHDB | [something indicating a recognized command]  |
|  | KEYS | [something indicating a recognized command]  |
|  | MONITOR | [something indicating a recognized command]  |
|  | PEXPIRE | [something indicating a recognized command]  |
|  | RENAME | [something indicating a recognized command]  |
|  | SAVE | [something indicating a recognized command]  |
|  | SHUTDOWN | [something indicating a recognized command]  |
|  | SPOP| [something indicating a recognized command]  |
|  | SREM| [something indicating a recognized command]  |
|  | SWAPDB| [something indicating a recognized command]  |

---
For background, see [this post about securing Redis](https://www.digitalocean.com/community/tutorials/how-to-secure-your-redis-installation-on-ubuntu-14-04#step-5-%E2%80%94-renaming-dangerous-commands), and refer to [this support inquiry from a cloud.gov customer](https://cloud-gov.zendesk.com/agent/tickets/1553).

**Comment by siennathesane** [comment]
I'd rather not go down this path right now, primarily because we are in the process of looking at other brokers, such as the aws-servicebroker, that can provide things like elasticache, which is still redis, so we should contribute there. on top of that, substitution ciphers are very easy to break, so even if all the commands were able to be renamed or prefixed, it doesn't really solve the problem.

this is extra work for not that much benefit when we already have higher-order priorities on the existing service brokers.

**Comment by bengerman13** [comment]
WRT the weak security - I *think* the user who requested this was thinking more about preventing accidental use, not malicious use.
There is also a feature of this rename wherein you rename the command to empty string, and nobody can use it at all.

**Comment by mogul** [comment]
Right, that was mentioned in that post. Can you think of a way of expressing that intent with parameters?

Mike, as for priority: This isn't something for us to focus on now that we know the customer isn't blocked by it, but I still want to capture user needs when we observe them. Priority is "out of scope" for the issue itself.
